sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/HiveThriftServer2Listener.scala

```diff
@@ -180,10 +180,12 @@ private[thriftserver] class HiveThriftServer2Listener(
   }
 
   private def onOperationClosed(e: SparkListenerThriftServerOperationClosed): Unit = {
-    executionList.get(e.id).closeTimestamp = e.closeTime
-    executionList.get(e.id).state = ExecutionState.CLOSED
-    updateStoreWithTriggerEnabled(executionList.get(e.id))
-    executionList.remove(e.id)
+    if (executionList.contains(e.id)) {
+      executionList.get(e.id).closeTimestamp = e.closeTime
+      executionList.get(e.id).state = ExecutionState.CLOSED
+      updateStoreWithTriggerEnabled(executionList.get(e.id))
+      executionList.remove(e.id)
+    }
```

Comment:
Just like this may be ok . 
If removed we don't need to do anything

Reply:
but I think this
```
    val view = kvstore.view(classOf[ExecutionInfo]).index("finishTime").first(0L)
    val toDelete = KVUtils.viewToSeq(view, countToDelete.toInt) { j =>
      j.finishTimestamp != 0
    }
```
can still delete executions that are FINISHED, but still fetching?

Reply:
> can still delete executions that are FINISHED, but still fetching?

https://github.com/apache/spark/blob/d38f8167483d4d79e8360f24a8c0bffd51460659/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/HiveThriftServer2AppStatusStore.scala#L113-L123

Here will control this. Won't remove `FINISHED` execution.

Reply:
:+1: thanks